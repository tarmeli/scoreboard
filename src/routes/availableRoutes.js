import Players from "../screens/players/Players";
import Home from "../screens/home/Home";
import { map } from "lodash/fp";

const routes = {
    home: {
        name: "Home",
        route: "/",
        screenComponent: Home,
        description: "Houston table tennis scoreboard",
    },

    players: {
        name: "Players",
        route: "/players",
        screenComponent: Players,
        description: "Players page",
    },
};

const availableRoutes = map((route) => ({ ...route }), routes);

export default availableRoutes;
