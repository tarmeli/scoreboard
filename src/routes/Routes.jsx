import React from "react";
import { Redirect, Route, Switch } from "react-router-dom";
import availableRoutes from "./availableRoutes";
import Map from "../components/Utility/Map/Map";

const Routes = () => (
    <Switch>
        <Map items={availableRoutes}>
            {({ screenComponent, screen: Screen = screenComponent, route }) => (
                <Route exact path={route}>
                    <Screen route={route} />
                </Route>
            )}
        </Map>

        <Redirect to="/" />
    </Switch>
);

export default Routes;
