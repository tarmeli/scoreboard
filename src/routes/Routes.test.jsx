import React from "react";
import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";

import availableRoutes from "./availableRoutes";

import Routes from "./Routes";

describe("Routes", () => {
    availableRoutes.forEach(({ name, route }) => {
        it(`renders ${name} screen`, () => {
            const component = mount(
                <MemoryRouter initialEntries={[route]}>
                    <Routes />
                </MemoryRouter>
            );

            expect(component.render()).toMatchSnapshot();
        });
    });
});
