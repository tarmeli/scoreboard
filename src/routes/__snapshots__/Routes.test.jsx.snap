// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes renders Home screen 1`] = `
<div>
  <div>
    <div />
  </div>
</div>
`;

exports[`Routes renders Players screen 1`] = `
<div>
  <div>
    <div>
      <div>
        <div>
          <table>
            <thead>
              <tr>
                <th>
                  Player name
                </th>
                <th />
                <th>
                  Wins
                </th>
                <th />
                <th />
                <th />
                <th>
                  Losses
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
          <form>
            <input
              value=""
            />
            <button
              disabled=""
              type="submit"
            >
              Add Player
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Routes renders default route 1`] = `
<div>
  <div>
    <div />
  </div>
</div>
`;
